test/sql/local/irc/test_create_secret_attach_combinations.test

```diff
@@ -0,0 +1,177 @@
+# name: test/sql/local/irc/test_create_secret_attach_combinations.test
+# description: test combinations of create secret and attah
+# group: [irc]
+
+require-env ICEBERG_SERVER_AVAILABLE
+
+require avro
+
+require parquet
+
+require iceberg
+
+require httpfs
+
+# Do not ignore 'HTTP' error messages!
+set ignore_error_messages
+
+# Test catalog secret and storage secret
+statement ok
+set enable_logging=true
+
+statement ok
+set logging_level='debug'
+
+statement ok
+CREATE SECRET iceberg_secret (
+	TYPE ICEBERG,
+	CLIENT_ID 'admin',
+	CLIENT_SECRET 'password',
+	OAUTH2_SERVER_URI 'http://127.0.0.1:8181/v1/oauth/tokens',
+	Authorization_type 'Oauth2'
+);
+
+statement ok
+CREATE SECRET storage_secret(
+	TYPE S3,
+	KEY_ID 'admin',
+	SECRET 'password',
+	ENDPOINT '127.0.0.1:9000',
+	URL_STYLE 'path',
+	USE_SSL 0
+);
+
+
+statement ok
+ATTACH '' AS my_datalake (
+	TYPE ICEBERG,
+	SECRET iceberg_secret,
+	ENDPOINT 'http://127.0.0.1:8181'
+);
+
+
+query I
+SELECT message FROM duckdb_logs where type='iceberg.Catalog.Curl.HTTPRequest' and message like '%POST%' order by timestamp
+----
+POST http://127.0.0.1:8181/v1/oauth/tokens (curl code 'No error')
+
+query I
+select count(*) from my_datalake.default.pyspark_iceberg_table_v2;
+----
+6592
+
+# drop the iceberg secret and put everything in the attach statement
+statement ok
+DROP SECRET iceberg_secret;
+
+statement ok
+DETACH my_datalake;
+
+statement ok
+ATTACH '' AS my_datalake (
+	TYPE ICEBERG,
+	CLIENT_ID 'admin',
+	CLIENT_SECRET 'password',
+	OAUTH2_SERVER_URI 'http://127.0.0.1:8181/v1/oauth/tokens',
+	ENDPOINT 'http://127.0.0.1:8181'
+);
+
+query I
+select count(*) from my_datalake.default.pyspark_iceberg_table_v2;
+----
+6592
+
+statement ok
+Detach my_datalake;
+
+# create bad storage secret to test storage secret
+statement ok
+CREATE SECRET storage_secret_2 (
```

Comment:
This create secret statement can be removed since it's not used anymore along with the drop 2 statements later